src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -68,6 +68,9 @@ private static bool PutExInternal(ApplicationEngine engine, StorageContext conte
 
             private static bool Storage_GetContext(ApplicationEngine engine)
             {
+                ExecutionContextState state = engine.CurrentContext.GetState<ExecutionContextState>();
+                if (!state.AllowModifyStates) return false;
```

Comment:
Modifying state refers not only to storage, but also many other SYSCALLs that modify the state. It is best to specify permissions when SYSCALL is registered.

Reply:
>  It is best to specify permissions when SYSCALL is registered.

Where?

Reply:
https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/InteropDescriptor.cs

Reply:
https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/InteropService.cs

Reply:
But Interops are registered as static way, and this is relative to the `ExecutionContext`